### PR TITLE
Feature/generation check for kubernetes update

### PR DESCRIPTION
**Pull request by Lafunamor**

### CODE REVIEW

File: broker/src/main/groovy/com/swisscom/cloud/sb/broker/services/kubernetes/facade/AbstractKubernetesFacade.groovy [hunk 1]
```diff
@@ -141,6 +144,29 @@ abstract class AbstractKubernetesFacade<T extends AbstractKubernetesServiceConfi
...
+    boolean isKubernetesUpdateSuccessful(String serviceInstanceGuid) {
+        try {
+            DeploymentList deploymentList = kubernetesClient.exchange(
+                    endpointMapperParamsDecorated.getEndpointUrlByTypeWithParams('Deployment', ["serviceInstanceGuid": serviceInstanceGuid]).getFirst(),
+                    HttpMethod.GET,
+                    "",
+                    DeploymentList.class
+            ).body
+            int numberOfDeployments = deploymentList.items.size()
+            int updatedDeployments = 0
```
Comment (meip) [inline broker/src/main/groovy/com/swisscom/cloud/sb/broker/services/kubernetes/facade/AbstractKubernetesFacade.groovy]: Shorter version: `int updatedDeployments = deploymentList.items.findAll(){ deployment -> deployment.status.updatedReplicas == deployment.status.replicas}.size()`

File: broker/src/main/groovy/com/swisscom/cloud/sb/broker/services/kubernetes/facade/redis/KubernetesFacadeRedis.groovy [hunk 2]
```diff
@@ -64,6 +64,23 @@ class KubernetesFacadeRedis extends AbstractKubernetesFacade<KubernetesRedisConf
         (new HashMap<String, String>(kubernetesServiceConfig.redisConfigurationDefaults)) << planBindings << serviceDetailBindings << otherBindings
     }
 
+    @Override
+    protected Map<String, String> getUpdateBindingMap(RequestWithParameters request) {
+        def serviceDetailBindings = getServiceDetailBindingMap(request)
+        def planBindings = getPlanParameterBindingMap(request.plan)
+        ServiceInstance serviceInstance = provisioningPersistenceService.getServiceInstance(request.serviceInstanceGuid)
+        def redisPassword = ServiceDetailsHelper.from(serviceInstance).getValue(KubernetesRedisServiceDetailKey.KUBERNETES_REDIS_PASSWORD)
```
Comment (meip) [inline broker/src/main/groovy/com/swisscom/cloud/sb/broker/services/kubernetes/facade/redis/KubernetesFacadeRedis.groovy]: Why are these values (`redisPassword`, `slaveofCommand` and `configCommand`) generated again for update?

File: broker/src/main/groovy/com/swisscom/cloud/sb/broker/services/kubernetes/client/rest/KubernetesClient.groovy [hunk 3]
```diff
@@ -75,12 +74,16 @@ class KubernetesClient<RESPONSE> {
             overrides[0]?.each { k, v ->
                 if (v instanceof Map && onto[k] instanceof Map)
                     merge((Map) onto[k], (Map) v)
+                else if (v instanceof List && onto[k] instanceof List)
```
[the remark this answers is not in the document]
Comment (Lafunamor) [reply broker/src/main/groovy/com/swisscom/cloud/sb/broker/services/kubernetes/client/rest/KubernetesClient.groovy]: added a comment that describes the functionality of this method

File: broker/src/main/groovy/com/swisscom/cloud/sb/broker/services/kubernetes/facade/redis/KubernetesFacadeRedis.groovy (hunk 2 above)
Comment (Lafunamor) [reply broker/src/main/groovy/com/swisscom/cloud/sb/broker/services/kubernetes/facade/redis/KubernetesFacadeRedis.groovy]: the redisPassword is gathered from the DB.
The slaveOfCommand and configCommand will be gathered from the API in the future (they are not saved to the DB). I've added stub implementations for the API calls.

**Pull request merged**